examples/GHZsetup/GHZsetup.jl

```diff
@@ -0,0 +1,92 @@
+using ResumableFunctions
+using ConcurrentSim
+using Revise
+using QuantumSavory
+using QuantumSavory.ProtocolZoo
+const bell = StabilizerState("XX ZZ")
+const perfect_pair = (Z1⊗Z1 + Z2⊗Z2) / sqrt(2)
+const perfect_pair_dm = SProjector(perfect_pair)
+const mixed_dm = MixedState(perfect_pair_dm)
+
+noisy_pair_func_depol(p) = p*perfect_pair_dm + (1-p)*mixed_dm
+
+function noisy_pair_func(F)
+    p = (4*F-1)/3
+    return noisy_pair_func_depol(p)
+end
+
+S = 5 # number of sensors
+F = 0.99 # fidelity
+
+noisy_pair = noisy_pair_func(F)
+
+@resumable function GHZ_projection(sim, net, S; time=0.1)
+    while true
+        queries = []
+        incomplete = false
+        # check & wait for all entanglements
+        for i in 1:S
+            q = query(net[i], EntanglementCounterpart, S + 1, ❓; locked=false, assigned=true)
+            if isnothing(q)
+                @yield timeout(sim, 0.1)
+                incomplete = true
+                break
+            end
+            push!(queries, q)
+        end
+        if incomplete
+            continue
+        end
+
+        @debug "All entanglements are ready, at $(now(sim))"
+
+        # GHZ -> computational basis
+        for i in 2:S
+            apply!([net[S + 1, 1], net[S + 1, i]], CNOT)
+        end
+        apply!(net[S + 1, 1], H)
+
+        # measure & send correction message
+        m1 = project_traceout!(net[S + 1, 1], Z)
+
+        if m1 == 2
+            msg1 = Tag(EntanglementUpdateX, S + 1, 1, 1, -1, -1, m1)
+            put!(channel(net, S + 1 => 1; permit_forward=true), msg1)
+        end
+
+        for i in 2:S
+            m = project_traceout!(net[S + 1, i], Z)
+            msg = Tag(EntanglementUpdateZ, S + 1, i, 1, -1, -1, m)
+            put!(channel(net, S + 1 => i; permit_forward=true), msg)
+        end
+
+        @yield timeout(sim, time)
+    end
+end
+
+
+# run the simulation
+
+
+net = RegisterNet(vcat([Register(1) for _ in 1:S], [Register(S)]))
```

Comment:
There is a slightly neater syntax for concatenation:
```
julia> [[1,2,3]; 1]
4-element Vector{Int64}:
 1
 2
 3
 1
```

or in this case `RegisterNet([[Register... for _ in ...]; Register(S)])`

Now that I typed this out I am actually unsure which one is more legible. Pick whichever you like best, and consider this comment just an FYI about the other syntax.